scripts/zkai_attention_kv_stwo_binary_typed_proof_accounting_gate.py

```
@@ -0,0 +1,685 @@
+                "record_stream_sha256": accounting["record_stream_sha256"],
+                "proof_sha256": row["proof_sha256"],
+            }
+        )
+    return out.getvalue()
+
+
+def validate_output_path(path: pathlib.Path) -> pathlib.Path:
+    path = path.resolve()
+    evidence_root = EVIDENCE_DIR.resolve()
+    if not path.parent.exists():
+        raise BinaryTypedProofAccountingGateError(f"output parent does not exist: {path.parent}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)